splice/static/js/controllers/distribution.js

```diff
@@ -0,0 +1,91 @@
+"use strict";
+
+angular.module('spliceApp').controller('distributionController', function($scope, spliceData, fileReader) {
+
+  $scope.distributions = null;
+  $scope.channels = null;
+  $scope.channelIndex = {};
+  $scope.choices = [];
+  $scope.payloadSchema = null;
+  $scope.channelSelect = null;
+  $scope.fileErrorMsg = null;
+  $scope.cache = {};
+  $scope.tiles = {};
+  $scope.downloadInProgress = false;
+
+  $scope.setupChannels = function(chans) {
+    $scope.channels = chans;
+    for (var i = 0; i < chans.length; i++) {
+      var chan = chans[i];
+      $scope.channelIndex[chan.id] = chan;
+    }
+  };
+
+  $scope.setupDistributions = function(dists) {
+    var choices = [];
+    $scope.distributions = dists;
+
+    if ($scope.channelSelect != null) {
+      // if already set, try to preserve the same selection
+      // while the values are the same, it isn't the same object
+
+      var oldId = $scope.channelSelect.id;
+      $scope.channelSelect = null;
+
+      for (var i = 0; i < $scope.channels.length; i++) {
+        if (oldId == $scope.channels[i].id) {
+          $scope.channelSelect = $scope.channels[i];
+          break;
+        }
+      }
+      // note that channelSelect could still be null after this
+    }
+
+    if ($scope.channelSelect == null) {
+      $scope.channelSelect = $scope.channels[0];
```

Comment:
There should always be channel data.

That said, I think you have a point. A more user-friendly error message will be shown in a coming commit.